src/BasePlatform.js

```diff
@@ -151,4 +151,44 @@ export default class BasePlatform {
     async setMinimizeToTrayEnabled(enabled: boolean): void {
         throw new Error("Unimplemented");
     }
+
+    supportsEventIndexing(): boolean {
```

Comment:
Suggestion: perhaps we can put all the methods here related to event indexing into an object returned from the platform? E.g. `platform.getEventIndexingManager` or whatever, returning null on unsupported platforms?

Reply:
Sure, one way to abstract it. Depends on if we want to declare the event plumbing (convert the MatrixEvent into a plain json object) platform specific to Seshat or if we want it to be more of a guidance for other indexer implementations ("hey your platform specific indexer needs to accept these types and return such a search result").